storage/url/url_test.go

```diff
@@ -432,3 +433,57 @@ func TestURLWithMode(t *testing.T) {
 		}
 	}
 }
+
+func TestURLSetRelative(t *testing.T) {
+	type testURLSetRelativeStruct struct {
+		name   string
+		base   string
+		target string
+		expect string
+	}
+	tests := make([]testURLSetRelativeStruct, 0, 10)
+	if runtime.GOOS != "windows" {
+		tests = append(tests, []testURLSetRelativeStruct{
+			// The expectation is the path which will get us to target starting
+			// from the directory of base object.
+			{"normal file path 1", "/parent/child/object", "/parent/child2/object", "../child2/object"},
```

Comment:
I believe it would make it more readable if field names are used when defining struct literals;
```suggestion
			{
				name:   "normal file path 1",
				base:   "/parent/child/object",
				target: "/parent/child2/object",
				expect: "../child2/object",
			},

```

Reply:
As for the case names, I think they should be more descriptive; instead of `normal file path 1` something like `local_sibling_child_object` etc.  IMHO it makes it is more understandable without need to check inputs/outputs to see what it actually tests.

Case names in other tests might give some ideas.